CRM/Member/BAO/Membership.php

```diff
@@ -2680,20 +2680,16 @@ public static function mergeMemberships($mainContactID, $otherContactID, &$sqlQu
    */
   public static function updateMembershipStatus($deceasedParams, $contactType) {
     $updateMembershipMsg = NULL;
-    $contactId = $deceasedParams['contact_id'] ?? NULL;
-    $deceasedDate = $deceasedParams['deceased_date'] ?? NULL;
+    $contactId = $deceasedParams['contact_id'];
+    $deceasedDate = $deceasedParams['deceased_date'];
 
     // process to set membership status to deceased for both active/inactive membership
     if ($contactId &&
       $contactType === 'Individual' &&
       !empty($deceasedParams['is_deceased'])
     ) {
 
-      $session = CRM_Core_Session::singleton();
-      $userId = $session->get('userID');
-      if (!$userId) {
-        $userId = $contactId;
-      }
+      $userId = CRM_Core_Session::getLoggedInContactID() ?? $contactId;
```

Comment:
@eileenmcnaughton I'm not 100% certain about this but there may be cases where `CRM_Core_Session::getLoggedInContactID()` returns the number 0 for anonymous. In that case we'd want to check if empty rather than isset as the previous 5 lines were doing. The equivalent one-liner to those five would be:

```suggestion
      $userId = CRM_Core_Session::getLoggedInContactID() ?: $contactId;
```

Reply:
Good point - fixed